jobs/bump-lockfile.Jenkinsfile

```diff
@@ -1,121 +1,124 @@
 
 properties([
-    pipelineTriggers([
-        // we don't need to bump lockfiles any more often than daily
-        cron("H H * * *")
+    // we're only triggered by bump-lockfiles
+    pipelineTriggers([]),
+    parameters([
+        choice(name: 'STREAM',
+               choices: streams.development,
+               description: 'Fedora CoreOS development stream to bump'),
     ])
 ])
 
-try { timeout(time: 120, unit: 'MINUTES') { cosaPod {
+echo "Waiting for bump-${params.STREAM} lock"
+currentBuild.description = "[${params.STREAM}] Waiting"
+
+try { lock(resource: "bump-${params.STREAM}") { timeout(time: 120, unit: 'MINUTES') { cosaPod {
+    currentBuild.description = "[${params.STREAM}] Running"
 
     // set up git user upfront
     shwrap("""
       git config --global user.name "CoreOS Bot"
       git config --global user.email "coreosbot@fedoraproject.org"
     """)
 
-    parallel branches.collectEntries { branch -> [branch, {
-        shwrap("mkdir ${branch}")
-        dir(branch) {
-            shwrap("cosa init --branch ${branch} https://github.com/${repo}")
-            shwrap("cosa buildprep ${BUILDS_BASE_HTTP_URL}/${branch}/builds")
-
-            def prevPkgChecksum = shwrapCapture("jq -c .packages src/config/manifest-lock.*.json | sha256sum")
-
-            // do a first fetch where we only fetch metadata; no point in
-            // importing RPMs if nothing actually changed
-            stage("Fetch Metadata") {
-                shwrap("cosa fetch --update-lockfile --dry-run")
-            }
-
-            def newPkgChecksum = shwrapCapture("jq -c .packages src/config/manifest-lock.*.json | sha256sum")
-            if (newPkgChecksum == prevPkgChecksum) {
-                println("No changes")
-                return
-            }
-
-            // sanity-check only base lockfiles were changed
-            shwrap("""
-              # do this separately so set -e kicks in if it fails
-              files=\$(git -C src/config ls-files --modified --deleted)
-              for f in \${files}; do
-                if ! [[ \${f} =~ ^manifest-lock\\.[0-9a-z_]+\\.json ]]; then
-                  echo "Unexpected modified file \${f}"
-                  exit 1
-                fi
-              done
-            """)
-
-            stage("Fetch") {
-                // XXX: hack around subtle lockfile bug (jlebon to submit an
-                // rpm-ostree issue or patch about this)
-                shwrap("cp src/config/fedora-coreos-pool.repo{,.bak}")
-                shwrap("echo cost=500 >> src/config/fedora-coreos-pool.repo")
-                shwrap("cosa fetch --strict")
-                shwrap("cp src/config/fedora-coreos-pool.repo{.bak,}")
-            }
-
-            stage("Build") {
-                shwrap("cosa build --strict")
-            }
-
-            // need to run this in the workspace context because `fcosKola()`
-            // archives directly there but archiveArtifacts is dir-sensitive
```

Comment:
```
// need to run this in the workspace context because `fcosKola()`
// archives directly there but archiveArtifacts is dir-sensitive
```

No longer applicable?

Reply:
Yeah, because now the workdir is directly in the workspace instead of a subdir.